src/lib/TypedEncoder.sol

```diff
@@ -222,16 +851,62 @@ library TypedEncoder {
 
         uint256 structsLen = chunk.structs.length;
         for (uint256 i = 0; i < structsLen; i++) {
-            bytes memory structEncoded = _encodeAbi(chunk.structs[i]);
+            Struct memory childStruct = chunk.structs[i];
+
+            bytes memory structEncoded;
+            if (childStruct.encodingType == EncodingType.Array) {
+                structEncoded = _encodeAsArray(childStruct);
+            } else if (childStruct.encodingType == EncodingType.CallWithSelector) {
+                structEncoded = _encodeCallWithSelector(childStruct);
+            } else if (childStruct.encodingType == EncodingType.CallWithSignature) {
+                structEncoded = _encodeCallWithSignature(childStruct);
+            } else if (childStruct.encodingType == EncodingType.Hash) {
+                // Hash encoding returns bytes32 (32 bytes)
+                structEncoded = abi.encodePacked(_encodeHash(childStruct));
+            } else if (childStruct.encodingType == EncodingType.Packed) {
+                // Packed encoding returns dynamic bytes
+                structEncoded = _encodePacked(childStruct);
+            } else if (childStruct.encodingType == EncodingType.Create) {
+                // Create encoding returns address (20 bytes)
+                structEncoded = abi.encodePacked(_encodeCreate(childStruct));
+            } else if (childStruct.encodingType == EncodingType.Create2) {
+                // Create2 encoding returns address (20 bytes)
+                structEncoded = abi.encodePacked(_encodeCreate2(childStruct));
+            } else if (childStruct.encodingType == EncodingType.Create3) {
+                // Create3 encoding returns address (20 bytes)
+                structEncoded = abi.encodePacked(_encodeCreate3(childStruct));
+            } else if (childStruct.encodingType == EncodingType.ABI) {
+                bytes memory innerEncoded = _encodeAbi(childStruct);
+                // Check if struct has dynamic field contents (not encoding type)
+                bool hasDynamicFields = _hasDynamicFields(childStruct);
+                structEncoded =
+                    hasDynamicFields ? abi.encodePacked(abi.encode(uint256(32)), innerEncoded) : innerEncoded;
```

Comment:
I though .ABI shouldn't have the offset but you are adding it here instead of below?